docs/cookbook/invoices/generating-invoice-after-payment.rst

```diff
@@ -0,0 +1,169 @@
+How to have the Invoice generated after the payment is paid?
+============================================================
+
+.. note::
+
+    This cookbook describes customization of a feature available only with `Sylius/InvoicingPlugin <https://github.com/Sylius/InvoicingPlugin/>`_ installed.
+
+The invoicing plugin lets you generate and download PDF invoices regarding orders. In it's default behavior the invoice
+is generated right after the customer creates and places the order.
+In this cookbook we will describe how to change this behavior so the invoice will be created right after the order would be paid.
+
+Why would you customize the invoice generation?
+-----------------------------------------------
+
+In the default case the order should be immutable, so any other data should not be changed. But let's say that your shop is customized with some of logic
+which is not out of the box. Maybe one of it will let you change the order after it is placed?
+In this case it would be better to have invoice generated after a particular step (in case of this cookbook - after the order is paid).
```

Comment:
```suggestion
In this case, it would be better to have an invoice generated after a particular step (in the case of this cookbook - after the order is paid).
```